src/main/client_context.cpp

```diff
@@ -232,11 +232,13 @@ std::unique_ptr<PreparedStatement> ClientContext::prepareNoLock(
     try {
         // parsing
         if (parsedStatement->getStatementType() != StatementType::TRANSACTION) {
-            auto txContext = this->transactionContext.get();
-            if (txContext->isAutoTransaction()) {
-                txContext->beginAutoTransaction(preparedStatement->readOnly);
+            if (transactionContext->isAutoTransaction()) {
+                if (transactionContext->hasActiveTransaction()) {
```

Comment:
Ditto explain the case causing this if statement. We will most likely change it later on.